### ISSUE TITLE
`__path__` is inferred an `Iterable[str]` rather than list

### ISSUE THREAD

**Issue by aDotInTheVoid** [microsoft/pyright#10058]
Note: This is a regression of #1032. I've confirmed it appears in the latest released version.

### Describe the bug

With the following code:

```
./mcve
└── __init__.py
```

`mcve/__init__.py`:

```python
print(__path__[0])
```

(Code sample in [pyright playground](https://pyright-play.net/?code=A4JwlgdgLgFA%2BnYBDKALBBtADAXQJRA))

Pyright give errors:

```
% pyright ./mcve 
/Users/aloenr01/tmp/mcve/__init__.py
  /Users/aloenr01/tmp/mcve/__init__.py:1:7 - error: "__getitem__" method not defined on type "Iterable[str]" (reportIndexIssue)
1 error, 0 warnings, 0 informations 
```

But the code works fine:

```
% python3
Python 3.12.9 (main, Feb  4 2025, 14:38:38) [Clang 16.0.0 (clang-1600.0.26.6)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import mcve
/Users/aloenr01/tmp/mcve
```

### VS Code extension or command-line

Command line: pyright 1.1.396

**Comment by erictraut** [comment]
In response to bug #1032, I changed this from `Iterable[str]` to `list[str]` back in 2020. In early 2021, I reverted this change for reasons I don't recall. Unfortunately, I can't find the bug report or question that prompted the revert.

The type `list[str]` is narrower than what is required by the runtime, so changing the type of `__path__` to `list[str]` would be incorrect. The type `Iterable[str]` is correct but probably wider than necessary. Based on the [typeshed stubs](https://github.com/python/typeshed/blob/28106bcb73835713231095f108f0994845092403/stdlib/types.pyi#L350), it appears that the correct type for this symbol is `MutableSequence[str]`. I'll change it to that.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.397.
